### DISCUSSION TITLE
Issues with New Install on Rpi4

### DISCUSSION THREAD

**Discussion by ghost** [mcguirepr89/BirdNET-Pi#339] [General]
Hello All-

had some issues using the all-in-one install script.   am now able to get the webserver working but most pages return php errors, including 

"tools"
```
Warning: Cannot modify header information - headers already sent by (output started at /home/pi/BirdNET-Pi/homepage/views.php:24) in /home/pi/BirdNET-Pi/homepage/views.php on line 107
```
```
Warning: Cannot modify header information - headers already sent by (output started at /home/pi/BirdNET-Pi/homepage/views.php:24) in /home/pi/BirdNET-Pi/homepage/views.php on line 108
You cannot edit the settings for this installation
```
"recordings"
```
Fatal error: Uncaught Error: Class 'SQLite3' not found in /home/pi/BirdNET-Pi/scripts/play.php:4 Stack trace: #0 /home/pi/BirdNET-Pi/homepage/views.php(137): include() #1 {main} thrown in /home/pi/BirdNET-Pi/scripts/play.php on line 4
```
Species stats page does load.

/var/log/syslog shows activity and not a lot of errors:

```
May 31 21:03:32 birdnetpi env[16242]: Line = 2022-05-31-birdnet-21:03:09.wav.csv
May 31 21:03:38 birdnetpi systemd[1]: chart_viewer.service: Scheduled restart job, restart counter is at 9.
May 31 21:03:38 birdnetpi systemd[1]: Stopped BirdNET-Pi Chart Viewer Service.
May 31 21:03:38 birdnetpi systemd[1]: chart_viewer.service: Consumed 9.302s CPU time.
May 31 21:03:38 birdnetpi systemd[1]: Started BirdNET-Pi Chart Viewer Service.
May 31 21:03:38 birdnetpi systemd[1]: spectrogram_viewer.service: Scheduled restart job, restart counter is at 117.
May 31 21:03:38 birdnetpi systemd[1]: Stopped BirdNET-Pi Spectrogram Viewer.
May 31 21:03:39 birdnetpi systemd[1]: Started BirdNET-Pi Spectrogram Viewer.
May 31 21:03:39 birdnetpi systemd[1]: spectrogram_viewer.service: Succeeded.
May 31 21:03:39 birdnetpi birdnet_analysis.sh[16195]: /home/pi/BirdNET-Pi/birdnet/bin/python3 /home/pi/BirdNET-Pi/scripts\analyze.py --i /home/pi/BirdSongs/May-2022/31-Tuesday/2022-05-31-birdnet-21:03:23.wav --o /home/pi/BirdSongs/May-2022/31-Tuesday/2022-05-31-birdnet-21:03:23.wav.csv --lat 47.6722 --lon -122.1257 --week 20 --overlap 0.0 --sensitivity 1.25 --min_conf 0.7
May 31 21:03:39 birdnetpi python3[5612]: ANALYZING AUDIO... DONE! Time 1.5 SECONDS
May 31 21:03:39 birdnetpi python3[5612]: WRITING RESULTS TO /home/pi/BirdSongs/May-2022/31-Tuesday/2022-05-31-birdnet-21:03:09.wav.csv ... DONE! WROTE 0 RESULTS.
May 31 21:03:39 birdnetpi python3[5612]: READING AUDIO DATA... DONE! READ 5 CHUNKS.
May 31 21:03:39 birdnetpi python3[5612]: Date-time: 2022-05-31 21:03:23
May 31 21:03:41 birdnetpi birdnet_analysis.sh[16315]: 0.0;3.0-('Salpinctes obsoletus_Rock Wren', 0.03811522)
May 31 21:03:41 birdnetpi birdnet_analysis.sh[16315]: 3.0;6.0-('Megascops kennicottii_Western Screech-Owl', 0.09126189)
May 31 21:03:41 birdnetpi birdnet_analysis.sh[16315]: 6.0;9.0-('Megascops kennicottii_Western Screech-Owl', 0.036578953)
May 31 21:03:41 birdnetpi birdnet_analysis.sh[16315]: 9.0;12.0-('Dendragapus obscurus_Dusky Grouse', 0.039875444)
May 31 21:03:41 birdnetpi birdnet_analysis.sh[16315]: 12.0;15.0-('Dendragapus obscurus_Dusky Grouse', 0.05871099)
May 31 21:03:41 birdnetpi systemd[1]: birdnet_analysis.service: Succeeded.
```

I did run `./scripts/update_birdnet.sh` as well.

Happy to troubleshoot but not sure where to start.

**Comment by avit** [comment]
I had a brief look at the `views.php` file and it's not clear to me how it's even supposed to work. 

1. Up top the page outputs HTTP headers with `session_start()`
2. Then some standard HTML content is output
3. Then it tries to output HTTP headers again, conditionally for the "Tools" page:

https://github.com/mcguirepr89/BirdNET-Pi/blob/main/homepage/views.php#L106-L110

I'm not sure how you get in this state, but sending response headers after outputting HTML is not valid.

I think if you can visit another page to get the HTTP Authentication prompt ("birdnet" / no password), and authenticate before visiting the tools page, then you might be able to get around this.

**Comment by ehpersonal38** [comment]
@avit You're right! Thankfully those headers that return 401 Unauthorized aren't actually needed as we "exit" a few lines below. So yeah, they should probably either be removed or reworked. Buttt those are just warnings anyway. 

Given that your install was a bit botched to begin with and you don't have any data at this point, I would advise just to do a complete reinstall.
`/usr/local/bin/uninstall.sh && cd ~ && rm -drf BirdNET-Pi`
`curl -s https://raw.githubusercontent.com/mcguirepr89/BirdNET-Pi/main/newinstaller.sh | bash`

Let me know if that doesn't work for you and I (or Patrick) will be happy to take another look. 🐤

**Comment by ghost** [comment]
Ok am trying again.    www.piwheels.org seems to be down atm, so that's holding things up.

**Comment by ghost** [comment]
Ok was able to uninstall and reinstall as directed above.   Still getting the same behavior

Seems to work -> Spectrogram, Species Stats

PHP errors -> Tools, Live Audio, Recordings, Daily Charts

**Reply by mcguirepr89** [reply]
Good morning, @davidcgregory -- thanks for your interest in installing BirdNET-Pi -- I'm sorry the installation is not currently working for you.

Each installation creates a log in the Raspberry Pi user's $HOME directory called `installation_xxxx.log`. If you could please provide that file here we will know why your installation is failing.

If www.piwheels.com is down, there won't be much (anything) that I can do until those repos become available again.

Let me know if you have any trouble retrieving your installation log ~~

My best regards,
Patrick
